explainers/permissions-policy-unload.md

```diff
@@ -178,6 +185,57 @@ This was rejected as it might cause unexpected behavior
 when frames communicate with each other.
 (We're open to suggestions)
 
+## Concerns about giving embedders control over the (non)execution of iframe code
+
+If the embedder of an iframe can control whether the iframe's unload handler runs or not
+then that seems to provide a cross-origin control point
+that could potentially be used to exploit the iframe.
+
+While cross-origin control of scripting is, in general, a bad thing,
+unload handlers are very different to other APIs.
+
+### Unload handlers already often don't run
+
+Right now unload handlers for an iframe will not be run in the following cases
+
+- mobile browser backgrounded tab killed by OS
+- most mobile browsers and some desktop browsers if they believe the page could be BFCached
+- any multi-process browser where a crash occurs in the process responsible for the iframe
+- implementations typically allot a time budget for running unload handlers
+  which may cause some not to complete or not to run at all
+
+This means that if preventing uload handlers from running is a problem
+then it's a problem that already exists.
+
+### Unload handlers are invisible to users
+
+Unload handlers run *after* navigation has occurred.
+The exact timing depends on many factors.
+In some cases unload handlers will block the navigation until they complete,
+in other cases they run in the background
+in parallel with the new page loading.
+In either case,
+whether the unload handler runs or not
+has no bearing on the immediate user experience
+apart from possibly slowing down the appearance of the next page.
+
+It is possible that skipping the unload handler will change the user experience later on,
+e.g. if unload is responsible for persisting some data
+in storage or over the network.
+However using unload for this is already a bad idea
+due to its unreliability.
+
+Since this policy requires opt-in by the embedder,
+problemative cases can be excepted until they are fixed.
+What remains are the cases where the embedder
+accidentally breaks something an iframe
+or (hypothetically) intentionally exploits the iframe.
+
+### Conclusion
+
+We believe that the danger posed by embedders maliciously disabling unload handlers in iframes is minimal
+and not in the same class as an ability to disable other APIs.
```

Comment:
Add something affirmative about how this trades off against allowing well-behaved outer frames to protect the user against bad experiences (e.g. slow-to-unload pages and broken bfcache), and how on balance we believe the small risks here are outweighed by the potential upside.

Reply:
Unfortunately anything that's slow to unload can just be slow to pagehide, they happen at the same timing and block the same things (in chrome at least). So I think bfcache is the win we can claim.

I've updated this.